src/Table/read_fits.cxx

```diff
@@ -70,77 +71,113 @@ void tablator::Table::read_fits(const boost::filesystem::path &path) {
     // If this table was created by write_fits(), those elements
     // would have been converted to labeled_properties and stored as keywords
-    // with values strings of the form
-    // "value XMLATTR_DOTatt1:val1 XMLATTR_DOTatt2:val2 ... XMLATTR_DOTattn:valn".
-    std::vector<std::pair<std::string, Property>> combined_labeled_properties;
-    for (auto &kwd : table_extension.keyWord()) {
-        std::string label(kwd.first), value;
+    // of the form
 
-        // Set to default and adjust as needed.
-        std::string name(label);
+    // ELEMENT.<prop_name>.XMLATTR.<attr_name> : attr_value or
+    // ELEMENT.<prop_name>.XMLATTR.ATTR_VALUE : prop.value
 
-        static boost::regex expr{"^(.*)" + DOT + XMLATTR_DOT + "(.*)$"};
-        boost::smatch what;
-        if (boost::regex_search(label, what, expr)) {
-            // Undo write_fits() hackery: extract name and label from
-            // "label.XMLATTR.name".
-            label.assign(what[1]);
-            name.assign(what[2]);
-        }
+    // where <prop_name> is the value of prop's ATTR_NAME attribute and is
+    // assumed to be non-empty for INFO elements, of which we might have many.
+    //(We can't yet convert to FITS format VOTables with more than one RESOURCE.
+    // 07Dec20)
 
-        if (std::find(fits_ignored_keywords.begin(), fits_ignored_keywords.end(),
-                      name) != fits_ignored_keywords.end()) {
-            continue;
-        }
+    std::vector<std::pair<std::string, Property>> combined_labeled_properties;
+    std::string prev_keyword = "";
+    std::string prev_label = "";
+    Property prop;
+
+    for (auto &kwd : table_extension.keyWord()) {
+        std::string keyword(kwd.first), value;
 
         // Annoyingly, CCfits does not have a way to just return the
         // value.  You have to give it something to put it in.
-        Property prop;
         std::string kwd_value;
         kwd.second->value(kwd_value);
-        std::vector<std::string> tokens;
-        boost::algorithm::split_regex(tokens, kwd_value, boost::regex(XMLATTR_DOT));
-        auto num_tokens = tokens.size();
-        for (size_t idx = 0; idx < num_tokens; ++idx) {
-            const auto &token = tokens.at(idx);
-            if (token.empty()) {
-                continue;
+
+        // If `kwd` was generated from a labeled_property by
+        // write_fits(), then the `keyword` string includes the value
+        // of ATTR_NAME for that property.  We group together all kwds
+        // with the same keyword and construct from them a
+        // labeled_property whose label is `label`.
+
+        // Set to defaults and adjust as needed.
+        std::string label(keyword);
+        std::string name(keyword);
+
+        // Used to undo write_fits() hackery.
+        static boost::regex attr_expr{"^(.*)" + DOT + XMLATTR_DOT + "(.*)$"};
+        static boost::regex info_expr{"^((?:.*\\.)?" + INFO + ")" + DOT + "(.*)$"};
```

Comment:
Shouldn't you escape the wildcard characters in DOT and XMLATTR_DOT (`"\\."` instead of `"."`)?

Reply:
The non-wildcard '.'s?  Yep.